### PR TITLE
Implement remaining methods in polaris db connector service with tests

### PR DESCRIPTION
**Pull request by rsyncv**

### CODE REVIEW

**Comment by raveeram** [comment]
few minor comments, nice work jeff!

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java [hunk 1]
```diff
@@ -57,8 +83,20 @@ public void delete(final ConnectorRequestContext requestContext, final Qualified
      */
     @Override
     public void update(final ConnectorRequestContext context, final DatabaseInfo databaseInfo) {
+        final QualifiedName databaseName = databaseInfo.getName();
         try {
-            // TODO: implement once data model supports this operation
```
[the remark this answers is not in the document]
Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java:61]: Good suggestion!

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java [hunk 2]
```diff
@@ -71,11 +109,93 @@ public void update(final ConnectorRequestContext context, final DatabaseInfo dat
...
-            return null;
+            final PolarisDatabaseMapper mapper = new PolarisDatabaseMapper(name.getCatalogName());
+            final Optional<PolarisDatabaseEntity> dbEntity = this.polarisConnector.getDatabase(name.getDatabaseName());
+            if (!dbEntity.isPresent()) {
+                throw new DatabaseNotFoundException(name);
+            }
+            return mapper.toInfo(dbEntity.get());
+        } catch (DatabaseNotFoundException exception) {
```
[the remark this answers is not in the document]
Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java:118]: Two reasons, one for logging reason, the other is if this specific exception is not caught and rethrown then the catch all exception block will catch and convert it to a more generic `ConnectorException`.

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java [hunk 3]
```diff
@@ -71,11 +109,93 @@ public void update(final ConnectorRequestContext context, final DatabaseInfo dat
...
+        @Nullable final QualifiedName prefix,
+        @Nullable final Sort sort,
+        @Nullable final Pageable pageable
+    ) {
+        try {
+            final List<DatabaseInfo> databaseInfos = Lists.newArrayList();
+            final PolarisDatabaseMapper mapper = new PolarisDatabaseMapper(name.getCatalogName());
+            for (PolarisDatabaseEntity entity : polarisConnector.getAllDatabases()) {
+                final QualifiedName qualifiedName = QualifiedName.ofDatabase(name.getCatalogName(), entity.getDbName());
+                if (prefix != null && !qualifiedName.toString().startsWith(prefix.toString())) {
```
[the remark this answers is not in the document]
Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java:187]: Ok, changed to do the check once.

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (raveeram) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java:116]: Ah I see, thanks, do we want to rethrow this in other methods above as well?

[the remark this answers is not in the document]
Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorDatabaseService.java:116]: Oh yeah let me add it to the `update` method too.

**Review by raveeram — APPROVED** [review]

**Pull request merged**